darts/metrics/metrics.py

```diff
@@ -6,43 +6,102 @@
 
+
 logger = get_logger(__name__)
 
 
+"""
+Note: for new metrics added to this module to be able to leverage the two decorators, it is required both having
+the `actual_series` and `pred_series` parameters, and not having other `Sequence` as args (since these decorators
+don't “unpack“ parameters different from `actual_series` and `pred_series`). In those cases, the new metric must take
+care of dealing with Sequence[TimeSeries] and multivariate TimeSeries on its own (See mase() implementation).
+"""
+
+
+def multi_ts_support(func):
+    """
+    This decorator further adapts the metrics that took as input two univariate/multivariate `TimeSeries` instances,
+    adding support for equally-sized array of `TimeSeries` instances. The decorator computes the pairwise metric for
+    `TimeSeries` with the same indices, and returns a float value that is computed as a function of all the
+    pairwise metrics using a `inter_reduction` subroutine passed as argument to the metric function.
+
+    If a 'Sequence[TimeSeries]' is passed as input, this decorator provides also parallelisation of the metric
+    evaluation regarding different `TimeSeries` (if the `n_jobs` parameter is not set 1).
+    """
+
+    @wraps(func)
+    def wrapper_multi_ts_support(*args, **kwargs):
+        actual_series = kwargs['actual_series'] if 'actual_series' in kwargs else args[0]
+        pred_series = kwargs['pred_series'] if 'pred_series' in kwargs else args[0] if 'actual_series' in kwargs \
+            else args[1]
+
+        n_jobs = kwargs.pop('n_jobs', 1)
+        verbose = kwargs.pop('verbose', False)
+
+        raise_if_not(isinstance(n_jobs, int), "n_jobs must be an integer")
+        raise_if_not(isinstance(verbose, bool), "verbose must be a bool")
+
+        actual_series = [actual_series] if not isinstance(actual_series, Sequence) else actual_series
+        pred_series = [pred_series] if not isinstance(pred_series, Sequence) else pred_series
+
+        raise_if_not(len(actual_series) == len(pred_series),
+                     "The two TimeSeries arrays must have the same length.", logger)
+
+        num_series_in_args = int('actual_series' not in kwargs) + int('pred_series' not in kwargs)
+        kwargs.pop('actual_series', 0)
+        kwargs.pop('pred_series', 0)
+
+        iterator = _build_tqdm_iterator(iterable=zip(actual_series, pred_series),
+                                        verbose=verbose,
+                                        total=len(actual_series))
+
+        value_list = _parallel_apply(iterator=iterator,
+                                     fn=func,
+                                     n_jobs=n_jobs,
+                                     fn_args=args[num_series_in_args:],
+                                     fn_kwargs=kwargs)
+
+        if 'inter_reduction' in kwargs:
+            return kwargs['inter_reduction'](value_list)
+        else:
+            return signature(func).parameters['inter_reduction'].default(value_list)
```

Comment:
Yep, I think it's a great idea. I would go for the second option, leaving the identity function as default. Maybe I can add the `np.mean` case as an example in the parameter's doc since it probably will be the most used one.